CMakeLists.txt

```diff
@@ -155,8 +155,8 @@ SET_TARGET_PROPERTIES(rmm PROPERTIES BUILD_RPATH "\$ORIGIN")
 target_include_directories(rmm PUBLIC
   "${THRUST_INCLUDE_DIR}"
```

Comment:
Should really move to using a target for this instead of CMake variables. This path likely won't resolve properly on the importing machine.

Reply:
@gigony's goal here is to have a consuming project use rmm via `add_subdirectory(rmm)`. For that purpose, I think this will work, since the consuming project will be built at the same time, so the include path will still be there.

You're right that this isn't going to work for a installed version of rmm -- which is my goal (in future PRs). thrust is going to be particularly hasslesome for this, since there may could be three ways of using thrust (from the cudatoolkit, installed elsewhere externally, or included via FetchContent. And in addition, `thrust` doesn't quite support the usual `find_package()` way of pulling it in :(

But for now, I believe this PR supports @gigony's use case and shouldn't break anything.

Reply:
> And in addition, `thrust` doesn't quite support the usual `find_package()` way of pulling it in :(

I believe this was changed as of Thrust 1.9.10, where there's now a nicer CMake target and the like: https://github.com/thrust/thrust/blob/main/thrust/cmake/README.md

Reply:
Yeah, I've seen that, but the "not quite the usual" thing is the `thrust_create_target(...)` (which does exist for good reasons, though).

I think we'll be able to work with it, but the thing about it is that a `find_package(rmm)` would pull in `thrust` and would have to do a `thrust_create_target()` internally, not exactly knowing what thrust variant the consuming application actually wants. It might not be a problem in the typical use case, but I think I can dream up issues where the application also makes a thrust target and one ends up pulling in thrust twice via different target names and with different configurations...